<?xml version="1.0"?>
<launch>
    <!-- <arg name="world_name" value="$(find social_navigation)/worlds/save2.world"/> -->
    <!-- <arg name="world_name" value="$(find social_navigation)/worlds/301.world"/> -->
    <arg name="world_name" value="$(find social_navigation)/worlds/301_actor.world"/>
    <arg name="use_sim_time" default="true" />
    <arg name="gui" default="true" />
    <arg name="headless" default="false" />

    <!-- robot pose -->
    <arg name="robot_x"     default="27.4" />
    <arg name="robot_y"     default="-0.5" />
    <arg name="robot_z"     default="0.07" />
    <arg name="robot_yaw"   default="3.14" />

    <!-- For hokuyo -->
    <arg name="hokuyo_laser" default="true" />
    <arg name="default_config" value="front_laser" if="$(arg hokuyo_laser)" />
    <!-- <arg name="default_config" value="base" unless="$(arg hokuyo_laser)" /> -->

    <!-- Configuration of Jackal which you would like to simulate.
        See jackal_description for details. -->
    <arg name="config" default="$(arg default_config)" />

    <!-- Optionally enable teleop for the simulation -->
    <arg name="joystick" default="false" />

    <!-- Launch Gazebo with the specified world -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="debug" value="0" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="use_sim_time" value="$(arg use_sim_time)" />
        <arg name="headless" value="$(arg headless)" />
        <arg name="world_name" value="$(arg world_name)" />
    </include>

    <!-- Spawn Jackal -->
    <!-- <include file="$(find jackal_gazebo)/launch/spawn_jackal.launch">
        <arg name="x" value="$(arg robot_x)" />
        <arg name="y" value="$(arg robot_y)" />
        <arg name="z" value="$(arg robot_z)" />
        <arg name="yaw" value="$(arg robot_yaw)" />
        <arg name="config" value="$(arg config)" />
        <arg name="joystick" value="$(arg joystick)" />
    </include> -->
    <param name="robot_description" command="cat $(find social_navigation)/config/jackal.urdf"/>

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model jackal -x $(arg robot_x) -y $(arg robot_y) -z $(arg robot_z) -R 0 -P 0 -Y $(arg robot_yaw)" />
    <include file="$(find jackal_control)/launch/control.launch" />

     <!--node pkg="social_navigation" type="global_planner" name="mcts" output="screen"/-->
</launch>
